src/event-stream/reader.ts

```diff
@@ -67,9 +75,53 @@ export function parseMessageTransactions(msg: CoreNodeBlockMessage): CoreNodeMes
     try {
       const txBuffer = Buffer.from(coreTx.raw_tx.substring(2), 'hex');
       const bufferReader = BufferReader.fromBuffer(txBuffer);
-      const rawTx = readTransaction(bufferReader);
-      const txSender = getTxSenderAddress(rawTx);
-      const sponsorAddress = getTxSponsorAddress(rawTx);
+      let txSender: string;
+      let sponsorAddress = undefined;
+      let rawTx: Transaction;
+      try {
+        rawTx = readTransaction(bufferReader);
+        txSender = getTxSenderAddress(rawTx);
+        sponsorAddress = getTxSponsorAddress(rawTx);
+      } catch (error) {
+        logError(
+          `BTC transaction found. Recreating from event. TX: ${JSON.stringify(coreTx)}: ${error}`,
+          error
+        );
+        const event = msg.events[i] as StxTransferEvent;
```

Comment:
should we instead check that we're looking at a "raw_tx = `0x00`" and only build that Tx if that's the case?

Reply:
Yep, probably!